File: zsh/export.py
from utils import get_cwd_path, get_system_config_path, get_home_path


def getRefs():
    cwd = get_cwd_path(__file__)
    config = get_system_config_path() / "zsh"

    return [
        (get_home_path() / ".zshenv", cwd / "zshenv"),
        (config / ".zshrc", cwd / "zshrc"),
        *[
            (config / s, cwd / s)
            for s in [
                "source",
                "plugins-config",
                "p10k-configs",
                "plugin-functions.sh",
                "zshrc.sh",
            ]
        ],
    ]

File: zsh/export.lua
local M = {}

M.links = function()
	local path = require("path")
	local zsh_config_dir = "~/.config/zsh"

	local links = {
		{
			"zshenv",
			"~/.zshenv",
		},
		{
			"zshrc",
			path(zsh_config_dir, ".zshrc"),
		},
	}

	local simple_refs = {
		"source",
		"plugins-config",
		"p10k-configs",
		"plugin-functions.sh",
		"zshrc.sh",
	}

	for _, f in ipairs(simple_refs) do
		table.insert(links, {
			f,
			path(zsh_config_dir, f),
		})
	end
	return links
end

return M
